runtime/runtime/tests/test_async_calls.rs

```diff
@@ -102,28 +100,26 @@ fn test_single_promise_no_callback() {
     let [r0] = &*assert_receipts!(group, signed_transaction) else {
         panic!("Incorrect number of produced receipts")
     };
-    let [r1, ref0] = &*assert_receipts!(group, "near_0" => r0 @ "near_1",
+    let receipts = &*assert_receipts!(group, "near_0" => r0 @ "near_1",
         ReceiptEnum::Action(ActionReceipt{actions, ..}), {},
         actions,
         a0, Action::FunctionCall(function_call_action),
         {
             assert_eq!(function_call_action.gas, GAS_1);
             assert_eq!(function_call_action.deposit, 0);
         }
-    ) else {
-        panic!("Incorrect number of produced receipts")
-    };
-    let [ref1] = &*assert_receipts!(group, "near_1" => r1 @ "near_2",
+    );
+    let [r1] = &receipts[..1] else { panic!("Incorrect number of produced receipts") };
```

Comment:
Style: consider as follows here and elsewhere:

```suggestion
    let [r1, ..] = &receipts;
```

Among other benefits, this keeps code more readable and avoids a fair bit of dead code.

Reply:
This is still a refutable pattern, since `let receipts = []` can't be matched to `let [r1, ..] = &receipts;` (what would `r1` be? . I couldn't find a way to avoid the `else panic!()` .

Reply:
I'd still get rid of `..1` index in favour of `..` in the pattern, but up to you.